Boolean_set_operations_2/include/CGAL/draw_polygon_set_2.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 1997
+// Utrecht University (The Netherlands),
+// ETH Zurich (Switzerland),
+// INRIA Sophia-Antipolis (France),
+// Max-Planck-Institute Saarbruecken (Germany),
+// and Tel-Aviv University (Israel).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Guillaume Damiand <guillaume.damiand@liris.cnrs.fr>
+
+#ifndef CGAL_DRAW_POLYGON_SET_2_H
+#define CGAL_DRAW_POLYGON_SET_2_H
+
+#include <CGAL/draw_polygon_with_holes_2.h>
+
+#ifdef DOXYGEN_RUNNING
+namespace CGAL {
+
+/*!
+\ingroup PkgDrawPolygonSet2
+
+opens a new window and draws `aps`, an instance of the `CGAL::Polygon_set_2` class. A call to this function is blocking, that is the program continues as soon as the user closes the window. This function requires CGAL_Qt5, and is only available if the flag CGAL_USE_BASIC_VIEWER is defined at compile time.
+\tparam PS an instance of the `CGAL::Polygon_set_2` class.
+\param aps the polygon set to draw.
+
+*/
+template<class PS>
+void draw(const PS& aps);
+
+} /* namespace CGAL */
+#endif
+
+#ifdef CGAL_USE_BASIC_VIEWER
+#include <CGAL/Polygon_set_2.h>
+
+namespace CGAL
+{
+
+template<class PS2>
+class SimplePolygonSet2ViewerQt :
+  public SimplePolygonWithHoles2ViewerQt<typename PS2::Polygon_with_holes_2>
+{
+  typedef SimplePolygonWithHoles2ViewerQt<typename PS2::Polygon_with_holes_2> Base;
+
+public:
+  SimplePolygonSet2ViewerQt(QWidget* parent, const PS2& aps2,
+                            const char* title="Basic Polygon_set_2 Viewer") :
+    Base(parent, title)
+  {
+    const int n = aps2.number_of_polygons_with_holes();
+    typename PS2::Polygon_with_holes_2 polygons[n];
```

Comment:
Except if I missed something n is not known at compile time, so you should replace the array with a `std::vector`.

```
         C:\cgal_root\CGAL-5.4-Ic-58\include\CGAL/draw_polygon_set_2.h(57,49): error C2131: expression did not evaluate to a constant [C:\cgal_root\CGAL-5.4-Ic-58\cmake\platforms\MSVC-2022-Preview-Release\test\Boolean_set_operations_2_Examples\draw_polygon_set.vcxproj]
```

Reply:
No, you're not missing anything.  I naively assumed that variable length arrays (a feature introduced to C in C99) were also supported in C++.  They are not, but GCC was being helpful in allowing their use.  I've instructed him to be sufficiently pedantic so as to avoid such problems in the future.

On the other hand GCC does not seem to mind the implicit `size_t` to `int` conversion even with `-Wextra` and `-Wpedantic`, but I've switched to using a `vector`, so the offending line was removed anyway.

Reply:
The variable length arrays is common known as a mistake that should have never been in C99. It has never entered the C++ norm. See for example that StackOverflow answer, for the reasons:
https://stackoverflow.com/a/21519062/1728537